ReactiveCocoa/Swift/UnidirectionalBinding.swift

```diff
@@ -0,0 +1,315 @@
+import enum Result.NoError
+
+infix operator <~ {
+	associativity right
+
+	// Binds tighter than assignment but looser than everything else
+	precedence 93
+}
+
+public protocol BindingConsumer: class {
+	associatedtype ValueType
+
+	/// The lifetime of `self`. The binding operator uses this to determine when
+	/// the binding should be teared down.
+	var lifetime: Lifetime { get }
+
+	/// Consume a value from the binding.
+	func consume(_ value: ValueType)
+
+	/// Binds a signal to a property, updating the property's value to the latest
+	/// value sent by the signal.
+	///
+	/// - note: The binding will automatically terminate when the property is
+	///         deinitialized, or when the signal sends a `completed` event.
+	///
+	/// ````
+	/// let property = MutableProperty(0)
+	/// let signal = Signal({ /* do some work after some time */ })
+	/// property <~ signal
+	/// ````
+	///
+	/// ````
+	/// let property = MutableProperty(0)
+	/// let signal = Signal({ /* do some work after some time */ })
+	/// let disposable = property <~ signal
+	/// ...
+	/// // Terminates binding before property dealloc or signal's
+	/// // `completed` event.
+	/// disposable.dispose()
+	/// ````
+	///
+	/// - parameters:
+	///   - consumer: A consumer to bind to.
+	///   - signal: A signal to bind.
+	///
+	/// - returns: A disposable that can be used to terminate binding before the
+	///            deinitialization of property or signal's `completed` event.
+	@discardableResult
+	static func <~ <Source: SignalProtocol where Source.Value == ValueType, Source.Error == NoError>(consumer: Self, signal: Source) -> Disposable
```

Comment:
`DynamicProperty` would override it, if we would match its semantics to the one of `Property`s.

Specifically, if we reverse the reference so that `DynamicProperty` now holds a strong reference to the `NSObject`, the `DynamicProperty` must be able to supply its own implementation of bindings that interacts directly with the underlying `NSObject`.